runners/spark/src/main/java/org/apache/beam/runners/spark/translation/SparkProcessContext.java

```diff
@@ -125,8 +164,50 @@ public InputT element() {
 
   @Override
   public void outputWithTimestamp(OutputT output, Instant timestamp) {
-    output(WindowedValue.of(output, timestamp,
-        windowedValue.getWindows(), windowedValue.getPane()));
+    if (windowedValue == null) {
+      // this is start/finishBundle.
+      output(noElementWindowedValue(output, timestamp, windowingStrategy));
+    } else {
+      output(WindowedValue.of(output, timestamp, windowedValue.getWindows(),
+          windowedValue.getPane()));
+    }
+  }
+
+  static <T> WindowedValue<T> noElementWindowedValue(final T output,
+                                                     final Instant timestamp,
+                                                     WindowingStrategy<?, ?> ws) {
+    @SuppressWarnings("unchecked")
+    WindowFn<Object, ?> windowFn = (WindowFn<Object, ?>) ws.getWindowFn();
```

Comment:
You're right! it's leftover from a mid-work state before I used a dedicated `SideInputReader`, I needed it for the sideInputs, don't need it anymore. I'll pass the `WindowFn` only.